lib/src/rules/parameter_assignments.dart

```diff
@@ -88,9 +136,33 @@ class _Visitor extends SimpleAstVisitor {
       // Getters don't have parameters.
       parameterList.parameters.forEach((e) {
         if (node.body.isPotentiallyMutatedInScope(e.element)) {
-          rule.reportLint(e);
+          _reportIfSimpleParameterOrWithDefaultValue(e, node);
         }
       });
     }
   }
+
+  void _reportIfSimpleParameterOrWithDefaultValue(
```

Comment:
This could be made more efficient by
(a) testing for simple parameters and default parameters with values before traversing any of the nodes, and
(b) using a for loop to traverse all of the nodes a single time and generating errors as you find them rather than creating a new list.